diff --git a/dialogs/u_changenode.cpp b/dialogs/u_changenode.cpp
index b6a2b36..4ccd69f 100644
--- a/dialogs/u_changenode.cpp
+++ b/dialogs/u_changenode.cpp
@@ -1,6 +1,7 @@
 #include "dialogs/u_changenode.h"
 #include "ui_u_changenode.h"
 #include "../control/messagebox.h"
+#include <cstring>
 #include <QHostInfo>
 #include <QUrl>
 
diff --git a/state/m_airdrop.h b/state/m_airdrop.h
index fd90c23..0f4c2bb 100644
--- a/state/m_airdrop.h
+++ b/state/m_airdrop.h
@@ -20,7 +20,7 @@ struct AirdropRequests {
     QString btcAddress;
     QString challendge;
     QString signature;
-    int64_t timestamp; // ms since epoch
+    long long timestamp; // ms since epoch
 
     void setData(const QString & btcAddress,
                  const QString & challendge,
diff --git a/util/Files.h b/util/Files.h
index a76fd0d..1e79d78 100644
--- a/util/Files.h
+++ b/util/Files.h
@@ -2,6 +2,7 @@
 #define MWC_QT_WALLET_FILES_H
 
 #include <QStringList>
+#include <functional>
 
 namespace util {
 
